src/test/java/com/icthh/xm/uaa/security/DomainTokenServicesUnitTest.java

```diff
@@ -155,6 +177,37 @@ public void testRefreshToken() throws Exception {
         assertEquals(USER_KEY, ((Map<String, String>) auth.getDetails()).get(AUTH_USER_KEY));
     }
 
+    @Test
+    public void testRefreshTokenForDeactivatedUser() throws Exception {
+        when(userService.findOneByLogin(LOGIN)).thenAnswer(invocation -> {
+            User user = new User();
+            user.setActivated(false); // user is deactivated
+            user.setUserKey(USER_KEY);
+            return Optional.of(user);
+        });
+
+        OAuth2AccessToken accessToken = tokenServices.createAccessToken(createAuthentication(LOGIN, TENANT, ROLE));
+
+        assertNotNull(accessToken);
+        assertNotNull(accessToken.getRefreshToken());
+        assertNotNull(accessToken.getRefreshToken().getValue());
+
+        Map<String, String> params = new HashMap<>();
+        params.put("grant_type", "refresh_token");
+        params.put("refresh_token", accessToken.getRefreshToken().getValue());
+
+        TokenRequest tokenRequest = new TokenRequest(params, CLIENT, null, "refresh_token");
+        try {
+            tokenServices.refreshAccessToken(accessToken.getRefreshToken().getValue(), tokenRequest);
+            fail("Expected UserNotActivatedException to be thrown");
+        } catch (UserNotActivatedException e) {
```

Comment:
Why we need this?

Reply:
@ishkurko This test ensures that deactivated user can't refresh his/her token. I might not understand your question correctly, so if my answer doesn't answer, please elaborate.

Reply:
Why we need this try catch with ignore comment

Reply:
@ishkurko here it is:

```
try {
            tokenServices.refreshAccessToken(accessToken.getRefreshToken().getValue(), tokenRequest);  <= call method which is expected to throw an exception which is instance of UserNotActivatedException.class
            fail("Expected UserNotActivatedException to be thrown");  <= in case we get here - it's an error, because we expect UserNotActivatedException to be thrown
        } catch (UserNotActivatedException e) {
      // in case we here - that's what we actually need
}
```

Reply:
Why not use JUnit native things like `@Test(expected=UserNotActivatedException.class)`?

Reply:
@ishkurko see the following conversation: https://github.com/xm-online/xm-uaa/pull/60#discussion_r289365623